src/context/ProductsContext.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  FC,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import axios from 'axios';
+import { Product } from '../types/Product';
+
+interface ProductsContextType {
+  products: Product[],
+  favourites: Product[],
+  toogleSelectFavorite: (product: Product) => void;
+  toogleSelectCart: (product: Product) => void;
+
+}
+const ProductsContext = createContext<ProductsContextType>({
+  products: [],
+  favourites: [],
+  toogleSelectFavorite: () => {},
+  toogleSelectCart: () => {},
+});
+
+export const useContextProvider = () => useContext(ProductsContext);
+
+type Props = {
+  children: React.ReactNode;
+};
+
+const apiURL = 'https://product-catalog-api-r8lb.onrender.com/products/';
+
```

Comment:
add here custom hook `export const useProductsContext = () => useContext(ProductsContext);`

this will allow us to use `useProductsContext` instead of  `useContext(ProductsContext)`